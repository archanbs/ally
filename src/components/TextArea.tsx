import React from "react";

export type ITextAreaProps = {
  label: string;
  id?: string;
  placeholder?: string;
  description?: string;
  textValue?: string;
  required?: boolean;
  onChange?: (val: string) => void;
};

export const TextArea = ({ label, id, placeholder, description, textValue, required, onChange }: ITextAreaProps) => (
  <div className="mt-4">
    <label className="font-semibold" htmlFor={id}>
      {label}
    </label>
    <br />
    <textarea
      id={id}
      name={id}
      className="border border-gray-400 px-1 rounded"
      rows={4}
      cols={40}
      placeholder={placeholder}
      required={required}
      value={textValue}
      onChange={(ev) => {
        ev.preventDefault();
        onChange(ev.target.innerText);
      }}
    />
    <br />
    <span className="text-[12px] text-[#4f4f4f]">{description}</span>
    <br />
    <br />
  </div>
);
