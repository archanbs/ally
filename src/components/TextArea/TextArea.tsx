import React from "react";

export type ITextAreaProps = {
  label: string;
  id?: string;
  placeholder?: string;
  description?: string;
  textValue?: string;
  required?: boolean;
  onChange?: (val: string) => void;
};

export const TextArea = ({ label, id, placeholder, description, textValue, required, onChange }: ITextAreaProps) => {
  const defaultId = id ?? label.split(" ").join("-");

  return (
    <div className="mt-4">
      <label className="font-semibold" htmlFor={defaultId}>
        {label}
      </label>
      <br />
      <textarea
        id={defaultId}
        name={defaultId}
        className="border border-gray-400 p-2 rounded text-[14px]"
        rows={4}
        cols={40}
        placeholder={placeholder}
        required={required}
        value={textValue}
        onChange={(ev) => {
          ev.preventDefault();
          onChange(ev.target.value);
        }}
      />
      <br />
      <span className="text-[12px] text-[#4f4f4f]">{description}</span>
      <br />
      <br />
    </div>
  );
};
