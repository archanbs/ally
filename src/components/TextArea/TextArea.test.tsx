import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { TextArea } from "./TextArea";

test("Text area is rendered correctly", () => {
  const compTree = renderer
    .create(<TextArea label="Mutliline text" placeholder="Enter multi-line text" description="Enter atleast 200 characters" required />)
    .toJSON();

  expect(compTree).toMatchSnapshot();
});

test("Text area onchange triggers the callback function", async () => {
  const obj = {
    changeCallback: function () {
      console.log("change callback called");
    },
  };
  jest.spyOn(obj, "changeCallback");

  render(
    <TextArea
      label="Mutliline text"
      placeholder="Enter multi-line text"
      description="Enter atleast 200 characters"
      required
      onChange={obj.changeCallback}
    />
  );
  const textArea = screen.getByLabelText("Mutliline text");
  expect(textArea).toBeInTheDocument();

  fireEvent.change(textArea, { target: { value: "multi-line text value" } });
  expect(obj.changeCallback).toHaveBeenCalled();
});
