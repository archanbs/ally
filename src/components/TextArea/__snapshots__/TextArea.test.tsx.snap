// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text area is rendered correctly 1`] = `
<div
  className="mt-4"
>
  <label
    className="font-semibold"
    htmlFor="Mutliline-text"
  >
    Mutliline text
  </label>
  <br />
  <textarea
    className="border border-gray-400 p-2 rounded text-[14px]"
    cols={40}
    id="Mutliline-text"
    name="Mutliline-text"
    onChange={[Function]}
    placeholder="Enter multi-line text"
    required={true}
    rows={4}
  />
  <br />
  <span
    className="text-[12px] text-[#4f4f4f]"
  >
    Enter atleast 200 characters
  </span>
  <br />
  <br />
</div>
`;
